src/main/java/org/terasology/managerInterface/nui/ManagementScreenLayer.java

```diff
@@ -0,0 +1,164 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.managerInterface.nui;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.Constants;
+import org.terasology.buildings.components.ConstructedBuildingComponent;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.prefab.PrefabManager;
+import org.terasology.logic.console.commandSystem.annotations.Command;
+import org.terasology.logic.console.commandSystem.annotations.CommandParam;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.managerInterface.managementBookComponent;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.managerInterface.VillageEntity;
+import org.terasology.rendering.nui.databinding.Binding;
+import org.terasology.rendering.nui.databinding.ReadOnlyBinding;
+import org.terasology.rendering.nui.widgets.UIButton;
+import org.terasology.rendering.nui.widgets.UILabel;
+import org.terasology.rendering.nui.widgets.UIList;
+import org.terasology.spawning.OreonSpawnComponent;
+import org.terasology.spawning.OreonSpawnEvent;
+
+import java.util.*;
+
+/**
+ * The screen which is triggered when player interacts({@code e press}) with a Management book block.
+ */
+public class ManagementScreenLayer extends CoreScreenLayer {
+    private static final Logger logger = LoggerFactory.getLogger(ManagementScreenLayer.class);
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private LocalPlayer localPlayer;
+
+    @In
+    private PrefabManager prefabManager;
+
+    private UIList ownedVillagesList;
+    private UIList managedVillagesList;
+    private UILabel selectedVillageNameLabel;
+    private UILabel selectedVillageInfoLabel;
+
+    private List<VillageEntity> debugOwnedVillages;
+    private List<VillageEntity> debugManagedVillages;
+
+
+    @Override
+    public void initialise() {
+        ownedVillagesList = find(Constants.OWNED_VILLAGES_LIST_ID, UIList.class);
+        managedVillagesList = find(Constants.MANAGED_VILLAGES_LIST_ID, UIList.class);
+        selectedVillageNameLabel = find(Constants.SELECTED_VILLAGE_NAME_LABEL_ID, UILabel.class);
+        selectedVillageInfoLabel = find(Constants.SELECTED_VILLAGE_INFO_LABEL_ID, UILabel.class);
+
+        List<String> cityNames = new ArrayList<String>();
```

Comment:
Check out the NameGenerator module. We could probably have a list of city names in there as well.